docs/book/src/topics/external-resource-gc.md

```diff
@@ -0,0 +1,29 @@
+# External Resource Garbage Collection
```

Comment:
It may be worth adding this doc link to https://cluster-api-aws.sigs.k8s.io/topics/bring-your-own-aws-infrastructure.html , wdyt?

Reply:
Good thinking....will change this :+1:

Reply:
Added